draft-ietf-mls-protocol.md

```diff
@@ -4836,6 +4846,93 @@ be securely removed from a group. It also allows a member to rotate their
 keypair such that the old private key can no longer be used to decrypt new
 messages.
 
+## Confidentiality of Group Metadata
+
+MLS does not provide confidentiality protection to some messages and fields
+within messages:
+
+* KeyPackage messages
+* GroupInfo messages
+* The unencrypted portion of a Welcome message
+* Any Proposal or Commit messages sent as MLSPlaintext messages
+* The unencrypted header fields in MLSCiphertext messages
+* The lengths of encrypted Welcome and MLSCiphertext messages
+
+The only mechanism MLS provides for confidentially distributing a group's
+ratchet tree to new members is to send it in a Welcome message as a
+`ratchet_tree` extension.  If an application distributes the tree in some other
+way, its security will depend on that application mechanism.
+
+A party observing these fields might be able to infer certain metadata about the
+group:
+
+* Group ID
+* Current epoch and frequency of epoch changes
+* Frequency of messages within an epoch
+* Group extensions
+* Group membership
+
+These fields are typically protected from parties other than the DS by applying
```

Comment:
This suggests that the DS is a separate party. As I noted in the other PR, that is not necessarily the case. The DS is an abstract component parts of which may run on the clients themselves. It makes sense to use the DS as adversary when it comes to metadata, but I think its abstract nature forces us to qualify a bit here.